File: src/java/foundgine-core/src/main/java/com/foundgine/core/semantic/resolution/SemanticReferenceEvidence.java
package com.foundgine.core.semantic.resolution;

import java.util.List;

public record SemanticReferenceEvidence(
    String query,
    List<RetrievalCandidate> candidates,
    double confidence,
    String interpretation) {
    public SemanticReferenceEvidence {
        candidates = candidates == null ? List.of() : List.copyOf(candidates);
    }
}

File: src/csharp/Foundgine.Core/Semantic/Resolution/SemanticReferenceEvidence.cs
namespace Foundgine.Core.Semantic.Resolution;

/// <summary>
/// Provider-neutral evidence used to ground natural-language references in the semantic model.
/// Retrieval providers produce evidence; the semantic layer decides whether it is sufficient to resolve.
/// </summary>
public sealed record SemanticReferenceEvidence(
    string Query,
    IReadOnlyList<RetrievalCandidate> Candidates,
    double Confidence,
    string Interpretation);
